app/src/main/java/com/poupa/vinylmusicplayer/preferences/SharedPreferencesImporter.java

```diff
@@ -0,0 +1,112 @@
+import android.os.Bundle;
+import android.os.ParcelFileDescriptor;
+import android.util.Log;
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import com.google.gson.ToNumberPolicy;
+import com.google.gson.ToNumberStrategy;
+import com.poupa.vinylmusicplayer.R;
+import com.poupa.vinylmusicplayer.ui.activities.SettingsActivity;
+import com.poupa.vinylmusicplayer.util.OopsHandler;
+import com.poupa.vinylmusicplayer.util.SafeToast;
+
+import java.io.BufferedReader;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.Map;
+
+import androidx.activity.result.ActivityResultLauncher;
+import androidx.activity.result.contract.ActivityResultContracts;
+import androidx.appcompat.app.AppCompatActivity;
+import androidx.preference.PreferenceManager;
+
+public class SharedPreferencesImporter extends AppCompatActivity {
+
+    private Context context;
+    private ActivityResultLauncher importFilePicker;
+
+    private SharedPreferences sharedPreferences;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        this.context = this.getApplicationContext();
+        sharedPreferences = PreferenceManager.getDefaultSharedPreferences(this.context);
+        importFilePicker = registerForActivityResult(new ActivityResultContracts.OpenDocument(), result -> {
+            // Unless the selection has been cancelled, create the export file
+            if(result != null) {
+                importSettings(result);
+            }
+            // Finishes the last activity to return to the settings activity.
+            this.finish();
+        });
+        importFilePicker.launch(new String[]{"text/plain"});
+    }
+
+    private void importSettings(Uri location) {
+        // Prepare the table used to store the lines
+        Map<String, ?> preferences;
+        Gson gson = new GsonBuilder().setObjectToNumberStrategy(ToNumberPolicy.LONG_OR_DOUBLE).create();
+        SharedPreferences.Editor spEditor = sharedPreferences.edit();
+
+        try {
+            // Try to open the file
+            ParcelFileDescriptor file = this.context.getContentResolver().openFileDescriptor(location, "r") ;
+            if(file == null) return;
+
+            // Read the content from the file line by line
+            BufferedReader reader = new BufferedReader(new FileReader(file.getFileDescriptor()));
+            preferences = gson.fromJson(reader, Map.class);
+
+            String fileFormat = (String) preferences.get("file_format");
+            int versionCode = (int) preferences.get("version_code");
+
+            if(fileFormat == "json" && versionCode == context.getPackageManager().getPackageInfo(context.getPackageName(), 0).versionCode) {
+
+                for (Map.Entry<String, ?> entry : preferences.entrySet()) {
+
+                    if(entry.getKey() == "file_format" || entry.getKey() == "version_code") continue;
```

Comment:
Please create new constants and use them/share them across classes, instead of string litterals.